src/bootstrap/format.rs

```diff
@@ -0,0 +1,28 @@
+//! Runs rustfmt on the repository.
+
+use crate::{util, Build};
+use std::process::Command;
+
+pub fn format(build: &Build, check: bool) {
+    let target = &build.build;
+    let rustfmt_path = build.config.initial_rustfmt.as_ref().unwrap_or_else(|| {
+        eprintln!("./x.py fmt is not supported on this channel");
+        std::process::exit(1);
+    }).clone();
+    let cargo_fmt_path = rustfmt_path.with_file_name(util::exe("cargo-fmt", &target));
+    assert!(cargo_fmt_path.is_file(), "{} not a file", cargo_fmt_path.display());
+
+    let mut cmd = Command::new(&cargo_fmt_path);
+    // cargo-fmt calls rustfmt as a bare command, so we need it to only find the correct one
+    cmd.env("PATH", cargo_fmt_path.parent().unwrap());
+    cmd.current_dir(&build.src);
+    cmd.arg("fmt");
+
+    if check {
+        cmd.arg("--");
+        cmd.arg("--check");
+    }
+
+    let status = cmd.status().expect("executing cargo-fmt");
```

Comment:
@anp IIUC you omit `cargo-fmt` output right in this line.

Reply:
From [Command::status](https://doc.rust-lang.org/std/process/struct.Command.html#method.status) docs:

> By default, stdin, stdout and stderr are inherited from the parent.

IIUC, this means that bootstrap's stdin/stdout/stderr are shared with the child process, which by default displays their output. This matches the behavior I see locally.

Maybe there's a way to test that rustfmt output is seen by the end user?

Reply:
I don't think we can do so in an automated fashion, but I presume you did check locally (on linux at least -- shouldn't be different I hope for windows)?

Reply:
I did check locally, yeah

Reply:
Alright, then we'll just wait for the retry and if that fails I guess retry again with some debugging inserted here.

Reply:
That way output is not synchronised and can sometimes act weird on Windows.

Reply:
Maybe it should be mentioned in docs?
https://github.com/rust-lang/rust/issues/67159 is another case when inherited streams don't work on Windows.